net/src/main/java/io/bitcoinsv/jcl/net/protocol/wrapper/P2P.java

```diff
@@ -147,6 +145,13 @@ private void init() {
                 }
             }
 
+            // share BlacklistHandler's blacklist tracker with P2PRequestHandler - both share the evidence of tracked peer blacklist counter(s)
```

Comment:
I understand the reason why you need to share the BlacklistTracker here, but I don't really think this is the best solution. If I get it right, the only reason why you share it is in case a "client" wants to create a _BlacklistPeerRequest_ without specifying the Duration, in that case you need this object to "calculate" that duration before you send the event to the Bus.

But this coupling between _P2P_, _P2PRequestBuilder_ and Stuff relevant only to the _BlackllistHandlerImpl_ is not good. 
I suggest the following:
- Remove the references to _BlacklistTracker_ from _P2P_ and _P2PRequestHandler_
- In _P2PRequestHandler_, when creating a _BlacklistPeerRequest_, if no reason is configured, then duration is set to NULL.
- In _BalcklistHandlerImpl_, every time we receive a _BlacklistPeerRequest_ with duration == NULL, we calculate it at that moment using _BlacllistTracker_.
- 
**Summary:**
 - Request to Blacklist a Peer without specifying duration, will set it to NULL.
 - duration == null means that the duration is calculated by Black listHandlerImpl once reception.
 - duration == Empty means infinite duration
 - duration == some Value, means we apply that duration

Reply:
The reason that `P2PRequestHandler` has the same `BlacklistTracker` instance is because if `BlacklistHandlerImpl` already blacklisted some hosts, blacklisting the same hosts via `P2PRequestHandler` should use the same counters to correctly determine which duration to use. But I can see how you wish to optimize this 👍
I will remove this reference and let `BlacklistHandlerImpl` do the calculation when receiving a `BlacklistPeerRequest`, we will just need then to keep in mind this handling of `null` and empty `Optional` values.